docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+S3OutputModule
+==============
+
+### About Amazon S3
+-------------------
+
+**Amazon S3 (Simple Storage Service)** is an object storage system with a web service interface to store and retrieve any amount of data at any time from anywhere on the web, offered by Amazon Web Services.
+
+### Overview
+------------
+
+We can upload files into S3 through partitioning the AbstractFileOutputOperator. But, S3OutputModule is needed to parallel uploading the blocks of a file into S3. 
+
+Below table illustrates the benefits of this module over AbstractFileOutputOperator.
+
+**S3OutputModule**|**AbstractFileOutputOperator**
+-----|-----
+Maximum upload file size is 5TB.|Maximum upload file size is 5GB.
+Best fit for both large and small files.|Best fit for small files.
+Module uses AmazonS3Client API's to upload objects into S3. Large files will upload using multipart feature and small files(Single Block) will upload using putObject(...) API|Operator uses Hadoop filesystems like S3AFileSystem.Consists of couple of steps to upload object into S3: (1) Write the data into the local filesystem. (2) When the stream closes, filesystem uploads the local object into S3.
+If a block fails to upload then you need to re-upload the data for that block only|If a file fails to upload then you need to re-upload the complete file.
+
+### About Multipart Upload Feature
+----------------------------------
+Uploading the blocks of a file into S3 can be achieved by [multipart feature](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).
+
+Using Multipart Upload feature, we can upload a single large file in multiple smaller parts. In Multipart Upload process, each part is a portion of object's data. Each part can be uploaded independently. 
+After all parts of a file are uploaded successfully, Amazon S3 combines the parts as a single object.
+
+Please refer the [Java code](http://docs.aws.amazon.com/AmazonS3/latest/dev/llJavaUploadFile.html) for uploading file into Amazon S3 bucket using multipart feature.
+
+### About S3 Output Module
+--------------------------
+
+Purpose of S3Output module is to upload files/directories into an Amazon S3 bucket using the multipart upload feature.
+
+S3Output module is **fault-tolerant**, **statically/dynamically partitionable** and has **exactly once** semantics.
+
+Module class is **S3OutputModule** located in the package **org.apache.apex.malhar.lib.fs.s3**; please refer to [github URL](https://github.com/apache/apex-malhar/blob/master/library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3OutputModule.java).
+
```

Comment:
Looks like this is the first module we are documenting and part of the reason it is hard to follow is that we don't have a document explaining what a module is. Could you add a short section here titled "Modules" and explain what they are and how they are similar to and yet are different from operators ?

Then, in the next section "Operators in S3OutputModule", include a diagram that shows what the actual DAG looks like for the same application i.e. show the internal operators and streams. That would provide the basis for the later discussion about parallel partitioning, etc.